components/kyma-operator/pkg/kymainstallation/step.go

```diff
@@ -60,7 +60,27 @@ func (s installStep) Run() error {
 		releaseOverrides)
 
 	if installErr != nil {
-		return errors.New("Helm install error: " + installErr.Error())
+		installErrMsg := fmt.Sprintf("Helm install error: %s ", installErr.Error())
+
+		isDeletable, err := s.helmClient.IsReleaseDeletable(s.component.GetReleaseName())
+		if err != nil {
+			errMsg := fmt.Sprintf("Checking status of %s failed with an error: %s", s.component.GetReleaseName(), err.Error())
+			log.Println(errMsg)
+			return errors.New(fmt.Sprintf("%s \n %s \n", installErrMsg, errMsg))
+		}
+
+		if isDeletable {
+			log.Println(fmt.Sprintf("Helm installation of %s failed. Deleting before retrying installation...", s.component.GetReleaseName()))
+			_, err := s.helmClient.DeleteRelease(s.component.GetReleaseName())
+
+			if err != nil {
+				deleteErrMsg := fmt.Sprintf("Helm delete of %s failed with an error: %s", s.component.GetReleaseName(), err.Error())
+				log.Println(deleteErrMsg)
+				return errors.New(fmt.Sprintf("%s \n %s \n", installErrMsg, deleteErrMsg))
+			}
+		}
+
+		return errors.New(installErrMsg)
```

Comment:
Maybe let's enhance the error message with info that installation failed but cleanup was done. This will less confusing during debugging